k8s/envoy-opa/README.md

```diff
@@ -0,0 +1,357 @@
+As a result, when you connect to the URL for the `frontend-2` service (e.g. `http://35.222.190.182:3002`), the browser only displays the title without any account details.
+
+![Frontend-2-no-details][frontend-2-view-no-details]
+
+[frontend-2-view-no-details]: images/frontend-2_view_no_details.png "Frontend-2 view no details account"
+
+After trying to display the `frontend-2` data, you can verify the decision made by OPA using the same `scripts/backend-opa-logs.sh` script as performed in the previous section. A similar log entry is available for the `frontend-2` service but with the result equal to `false` due to the SPIFFE ID mismatch.
+
+## Retesting frontend-2 with a New Policy
+
+Let's update the Rego policy to match the SPIFFE ID of the `frontend-2` and test again. There is a Bash script that we can leverage to complete this task. Once executed, it will open the editor defined by your `KUBE_EDITOR`, or `EDITOR` environment variables, or fall back to `vi` for Linux or Notepad for Windows.
+
+```console
+$ bash scripts/backend-update-policy.sh
+```
+
+With the editor open, look for the following line that specifies the SPIFFE ID:
+
+```console
+svc_spiffe_id == "spiffe://example.org/ns/default/sa/default/frontend"
+```
+
+Update that line to match the SPIFFE ID for the `frontend-2` workload:
+
+```console
+svc_spiffe_id == "spiffe://example.org/ns/default/sa/default/frontend-2"
+```
+
+Save the changes and exit. The `backend-update-policy.sh` script resumes. The script applies the new version of the ConfigMap and then restarts the `backend` pod to pickup the new rule.
+Wait some seconds for the deployment to propagate before trying to view the `frontend-2` service in your browser again.
+Once the pod is ready, refresh the browser using the correct URL for the `frontend-2` service (e.g. `http://35.222.190.182:3002`). As a result, now the page shows the account details for user _Alex Fergus_.
+
+![Frontend-2][frontend-2-view]
+
+[frontend-2-view]: images/frontend-2_view.png "Frontend-2 view"
+
+On the other hand, if you now connect to the URL for the `frontend` service (e.g. `http://35.222.164.221:3000`), the browser only displays the title without any account details. This is expected because the policy was updated and now the SPIFFE ID for the `frontend` service does not satisfy the policy anymore.
+
+# Cleanup
+
+When you are finished, you can use the following commands to clean the environment created for the tutorial. It will remove:
+   * All resources created for this SPIRE - Envoy with OPA integration tutorial.
+   * All resources created for the SPIRE - Envoy X.509 integration tutorial.
+   * All deployments and configurations for the SPIRE Agent, SPIRE Server, and namespace.
```

Comment:
Please remove the bullets at the end of these lines since they aren't full sentences.

```suggestion
   * All resources created for this SPIRE - Envoy with OPA integration tutorial
   * All resources created for the SPIRE - Envoy X.509 integration tutorial
   * All deployments and configurations for the SPIRE Agent, SPIRE Server, and namespace
```

Reply:
Ups, I fixed this for others tutorials and I forgot to do it here too. Thanks.